### PR TITLE
8326503: [11u] java/net/HttpURLConnection/HttpURLConnectionExpectContinueTest.java fail because of package org.junit.jupiter.api does not exist

### PR DESCRIPTION
**Pull request by antbob**
This patch addresses the test failures described in https://bugs.openjdk.org/browse/JDK-8326503 due to junit 5 package org.junit.jupiter.api which does not exist in older junit 4 that is being used here now for 11u builds by refactoring junit 5 api use to available junit 4 api.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8326503](https://bugs.openjdk.org/browse/JDK-8326503) needs maintainer approval

### Issue
 * [JDK-8326503](https://bugs.openjdk.org/browse/JDK-8326503): [11u] java/net/HttpURLConnection/HttpURLConnectionExpectContinueTest.java fail because of package org.junit.jupiter.api does not exist (**Bug** - P4 - Approved)

### Reviewers
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev.git pull/2553/head:pull/2553` \
`$ git checkout pull/2553`

Update a local copy of the PR: \
`$ git checkout pull/2553` \
`$ git pull https://git.openjdk.org/jdk11u-dev.git pull/2553/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2553`

View PR using the GUI difftool: \
`$ git pr show -t 2553`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/2553.diff">https://git.openjdk.org/jdk11u-dev/pull/2553.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk11u-dev/pull/2553#issuecomment-1963821842)

### CODE REVIEW

File: test/jdk/java/net/HttpURLConnection/HttpURLConnectionExpectContinueTest.java [hunk 1]
```diff
@@ -67,7 +65,7 @@ class Control {
         Logger.getLogger("").getHandlers()[0].setLevel(Level.ALL);
     }
 
-    @BeforeAll
+    @Before
```
Comment (jerboaa) [inline test/jdk/java/net/HttpURLConnection/HttpURLConnectionExpectContinueTest.java:68]: Wouldn't `@BeforeClass` be appropriate here?

File: test/jdk/java/net/HttpURLConnection/HttpURLConnectionExpectContinueTest.java [hunk 2]
```diff
@@ -170,7 +168,7 @@ public void startServerSocket() throws Exception {
         serverThread.start();
     }
 
-    @AfterAll
+    @After
```
Comment (jerboaa) [inline test/jdk/java/net/HttpURLConnection/HttpURLConnectionExpectContinueTest.java:171]: Same here: `@AfterClass`?

**Review by jerboaa — COMMENTED** [review]
Have you considered `@BeforeClass` and `@AfterClass` annotations?

**Comment by antbob** [comment]
> Have you considered `@BeforeClass` and `@AfterClass` annotations?

yes, but they require related methods to be static which would require further refactoring which does not worth a stretch in this case as the tests are fairly lightweight and quick

**Comment by jerboaa** [comment]
> > Have you considered `@BeforeClass` and `@AfterClass` annotations?
> 
> yes, but they require related methods to be static which would require further refactoring which does not worth a stretch in this case as the tests are fairly lightweight and quick

I think we should try. It looks like making `control` and `serverThread` `static` would be fine and be similar what the test in JDK 17 does.

**Comment by antbob** [comment]
refactored further with before/after class. while looking into this test some more it actually has concurrency problems with junit 5 and parallel execution (tho i think the parallel execution can only be explicitly enabled on demand) where Control would be shared by several tests. With junit 4 the tests are always sequential and never parallel so this concern is void but this is something to note here.

**Review by jerboaa — APPROVED** [review]
Looks fine to me.

**Comment by antbob** [comment]
/approval request this needs to land asap to avoid test failures with older jtreg 6.1 in 11u

**Pull request closed without merging**